sdk/communication/azure-communication-chat/src/main/java/com/azure/android/communication/chat/ChatAsyncClient.java

```diff
@@ -315,15 +315,14 @@ public void stopRealtimeNotifications() {
      * If there's an error during registration initialization or renewal,
      * push notifications will be disabled and {@code errorHandler} will be called.
      * @param deviceRegistrationToken Device registration token obtained from the FCM SDK.
-     * @param errorHandler error handler callback for registration failures
      * @throws RuntimeException if push notifications failed to start.
      */
-    public void startPushNotifications(String deviceRegistrationToken, Consumer<Throwable> errorHandler) {
+    public void startPushNotifications(String deviceRegistrationToken) {
```

Comment:
We are passing this handler to make the exception being handled inside the worker. The logic in worker is guaranteed to execute. Otherwise, we are using listener in pushnotification client to handle exception. The listener is not ensured to execute when the thread is idle/killed.

Reply:
Based on a separate discussion, we will be keeping this API around but will eventually deprecate it.